### DISCUSSION TITLE
Severe input lag on Pen + 2in1 labtop device

### DISCUSSION THREAD

**Discussion by minecraftfen** [ppy/osu#28455] [Q&A] [answered]
### Type

Game behaviour

### Bug description

In the video I use the relax mod and follow the notes normally with my pen, but the cursor always appears where my pen was seconds ago, and even after a failure I immediately remove the pen and the cursor keeps moving on its own.

---

osu! doesn't find any tablet, and my pen is processed as a mouse.
but when I use a mouse with relax mod, I got 100.0% Accuracy.
![image](https://github.com/ppy/osu/assets/42016099/d9686adb-1e59-4d39-8f9a-ac871ab00ccc)
![image](https://github.com/ppy/osu/assets/42016099/84c94664-cc17-497d-9832-6c5418a2dc92)
I closed high precision mouse because it cause some weird problems on my pen.

---

Hardware:
Huawei Matebook E 2023
M-Pencil (Second Generation)

---

I think these info I provided might not enough to indicates the problem that I experienced.
If any further info needed, please tell me.

### Screenshots or videos

https://github.com/ppy/osu/assets/42016099/2716b38c-3a65-4a0f-ad32-d5c8947199ac

### Version

2024.521.2-lazer

### Logs

[compressed-logs.zip](https://github.com/user-attachments/files/15799041/compressed-logs.zip)
This is not a log exported immediately after recording the video, I exported this log immediately after launching the game and playing the same beatmap with the same input method as in the video and encountering the same issues

**Comment by minecraftfen** [comment]
OS:
![image](https://github.com/ppy/osu/assets/42016099/1b3af622-7db8-4cd1-acb4-907aa880d719)
Hardware:
CPU: 12th Gen InteI(R) Core(TM) i7-1260U
RAM: 16G (No brand info found)
GPU: Intel(R) Iris(R) Xe Graphics

**Comment by bdach** [comment]
Please upload video of this with frame graph showing (ctrl-f11 twice).

**Reply by minecraftfen** [reply]
okey wait a min

**Reply by minecraftfen** [reply]
https://github.com/ppy/osu/assets/42016099/652b0250-3813-4832-bb6b-e817e144f0c5

**Reply by minecraftfen** [reply]
https://github.com/ppy/osu/assets/42016099/eb2e63cf-e438-4e3e-98c4-082ad16ec57f

**Reply by minecraftfen** [reply]
and in this video, I enabled high precision mouse and tried to click on a beatmap with my pen, but every time I click, it scroll.

**Reply by minecraftfen** [reply]
Then I tried to play a beatmap in the same video and the problem make itself a little differently, but it's still there

**Comment by minecraftfen** [comment]
@bdach R U here? I have done uploading

**Reply by bdach** [reply]
Yes you do not need to ping. I have other things to do besides this.

Looks like either a duplicate of https://github.com/ppy/osu-framework/issues/6181, or something on your system is absolutely abusing input hooks and slowing down mouse across the board for the entire system. Just for reference, are you only getting this in osu!, or is this a problem in other games/apps too?

**Reply by minecraftfen** [reply]
sorry for that
I'm testing about it.
I've found that edge browser, Adobe softwares, and whiteboards got no lag, but when I'm draging windows, I got some simular problems.

**Reply by minecraftfen** [reply]
Hmm, in the other words, may be when my pen is considered as a pen, it works normal.
but when it is considered as a mouse, it work with lags.

**Reply by minecraftfen** [reply]
I've searched for keywords like lag before but I didn't find it at that time
And I have used this pen for a year and I doesn't got any other problems so I consider it as a problem of osu

**Reply by minecraftfen** [reply]
So I opened that issue and thought I should be reporting a new bug in osu

**Reply by minecraftfen** [reply]
btw, is the Surface Pen, M-Pencil, or whatever, pens for 2-in-1 laptops, within osu's maintenance?

**Reply by minecraftfen** [reply]
If not, I think I should close this discussion.

**Reply by bdach** [reply]
You could try and see if closing apps you have open in the background, if any, helps alleviate this.

> btw, is the Surface Pen, M-Pencil, or whatever, pens for 2-in-1 laptops, within osu's maintenance?

We don't go out of our way to support these specifically. If it works it's mostly thanks to our upstreams (sdl in particular).

**Reply by minecraftfen** [reply]
Then go to the Windows Feedback Center and report a bug

**Reply by minecraftfen** [reply]
> You could try and see if closing apps you have open in the background, if any, helps alleviate this.
> 
> > btw, is the Surface Pen, M-Pencil, or whatever, pens for 2-in-1 laptops, within osu's maintenance?
> 
> We don't go out of our way to support these specifically. If it works it's mostly thanks to our upstreams (sdl in particular).

I've rebooted my device before screen recording.

**Reply by bdach** [reply]
Reboot is one thing but this may also be an effect of persistent background apps. Something like a discord, a parsec, some other app that just sits there in the system tray looking like it's doing nothing.

**Reply by minecraftfen** [reply]
got it, I will test it another time

**Reply by minecraftfen** [reply]
![image](https://github.com/ppy/osu/assets/42016099/01dcb052-ee32-4227-949a-3dc7d9fbcef2)
I rebooted another time, cleared system tray, and killed Adobe/Logitech updater and some other stuff in task mgr.

**Reply by minecraftfen** [reply]
99.53%, lag still exist, but much better
![image](https://github.com/ppy/osu/assets/42016099/666721d8-74f3-4854-9a14-ce6c24f4aecc)

**Reply by minecraftfen** [reply]
Seems like some thing in tray caused this

**Reply by minecraftfen** [reply]
Thanks, since this is no longer a problem of osu, I will close this discussion.
Sorry to interrupt.

**Reply by minecraftfen** [reply]
wait, it seems like lag still increase by time.

**Reply by minecraftfen** [reply, marked as answer]
okay I found it, [Microsoft Powertoys](https://apps.microsoft.com/detail/xp89dcgq3k6vld?hl=zh-cn&gl=HK) caused the problem.

**Reply by minecraftfen** [reply]
but I don't know why

**Reply by minecraftfen** [reply]
@bdach is there any "known incompatible apps" list? This app has a pretty big user base, I think it's needed to add it into a list like this.

**Reply by minecraftfen** [reply]
It's been almost an hour since you replied, and just in case you didn't notice due to your attention being diverted to other issues.

**Reply by BlockHead007** [reply]
> okay I found it, [Microsoft Powertoys](https://apps.microsoft.com/detail/xp89dcgq3k6vld?hl=zh-cn&gl=HK) caused the problem.

It's not the first time people reporting that Microsoft Powertoys can cause this issue (see #26822). What plugins are you using? I do use it as well, but I never got issues like this.

Also,
> It's been almost an hour since you replied

As he said, he has other things to do besides answering questions. Just wait for his reply.

**Reply by minecraftfen** [reply]
most of them
![image](https://github.com/ppy/osu/assets/42016099/7ef8b05b-4c0a-4477-b949-64b79a8096aa)
![image](https://github.com/ppy/osu/assets/42016099/82b8ffe4-7ff2-4d66-a64c-63cf34600e0a)

**Reply by minecraftfen** [reply]
I checked the DOM and the image exists, but I can't load it here. Can you see it there?

**Comment by minecraftfen** [comment]
I tried turning off mouse utils which was most likely causing the problem, but the problem still persisted.
